controllers/cloudstackfailuredomain_controller.go

```diff
@@ -42,6 +42,13 @@ type CloudStackFailureDomainReconciler struct {
 	csCtrlrUtils.ReconcilerBase
 }
 
+const (
+	conditionTypeControlPlaneReady = "ControlPlaneReady"
```

Comment:
nit: would we be able to use the constants from https://github.com/kubernetes-sigs/cluster-api/blob/6b5afa3d141da106ce03d163ebbe10a283b24d62/api/v1beta1/condition_consts.go#L78 and https://github.com/kubernetes-sigs/cluster-api/blob/6b5afa3d141da106ce03d163ebbe10a283b24d62/api/v1beta1/condition_consts.go#L24